packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -267,7 +267,9 @@ export function useInnerBlocksProps( props = {}, options = {} ) {
 
 	const ref = useMergeRefs( [
 		props.ref,
-		__unstableDisableDropZone || isDropZoneDisabled
+		__unstableDisableDropZone ||
+		isDropZoneDisabled ||
+		( layout?.columnCount && window.__experimentalEnableGridInteractivity )
 			? null
 			: blockDropZoneRef,
```

Comment:
Bit difficult to understand what's going on here. Might be good to split the logic pretending the `?` into a variable e.g. `isDropZoneEnabled`.